### PR TITLE
infra: setting up code infrastructure

### PR DESCRIPTION
**Pull request by devgupta2607**
**Motivation**:
Setting up code infrastructure so that it will be maintainable in future.

**Other information**
 Commit made by @krishnarai30 is my mistake. Bhaiya early login in using my computer.

### CODE REVIEW

**Comment by devgupta2607** [comment]
Restructured code commit is correct otherwise just person change but later corrected.

File: lib/src/screens/homepage.dart [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import 'package:flutter/material.dart';
+import 'dart:convert';
+
+Future getSpecieData(String name, BuildContext context) async {
```
Comment (rishabh3112) [inline lib/src/screens/homepage.dart]: Move this logic to a separate file, which will act as a data source for our app.

**Review by rishabh3112 — CHANGES_REQUESTED** [review]
Great start @devgupta2607 ✌

P.S.
If @krishnarai30 is not the person that has worked on this, please rewrite the git history to make yourself the author of the commit under suspicion.

File: lib/src/screens/homepage.dart (hunk 1 above)
Comment (devgupta2607) [reply lib/src/screens/homepage.dart]: ok I will look into this @rishabh3112

Comment (devgupta2607) [reply lib/src/screens/homepage.dart]: This will definately shift to a new file when cards for each animal will be created so no worries there

Comment (rishabh3112) [reply lib/src/screens/homepage.dart]: Okay!
I was lookin' to implement routes using [fluro](https://github.com/theyakka/fluro). WDYT?

**Pull request merged**